src/CcdImage.cc

```diff
@@ -146,65 +139,36 @@ CcdImage::CcdImage(lsst::afw::table::SortedCatalogT<lsst::afw::table::SourceReco
     TP2CTP = GtransfoCompose(&raDec2CTP, &TP2RaDec);
     sky2TP = new TanRaDec2Pix(identity, tanWcs->TangentPoint());
 
-      // this one is needed for matches :
+    // this one is needed for matches :
     pix2CommonTangentPlane = GtransfoCompose(&raDec2CTP, tanWcs);
 
-    // In the following we read informations directly from the fits header which is instrument dependent
-    // We rely on the camera name which is not optimal as a camera can be mounted on different telescopes.
-    // We would rather need the telescope name.
-    // This instrument specific part will eventually have to be handled by the camera mapper
-
-    double latitude;
-    double lst_obs;
-    double ra;
-    double dec;
-    if (camera == "megacam" || camera == "CFHT MegaCam") {
-        airMass = meta->get<double>("AIRMASS");
-        jd = meta->get<double>("MJD-OBS");  // Julian date
-        expTime = meta->get<double>("EXPTIME");
-        latitude = lsst::afw::geom::degToRad(meta->get<double>("LATITUDE"));
-        lst_obs = lsst::afw::geom::degToRad(RaStringToDeg(meta->get<std::string>("LST-OBS")));
-        ra = lsst::afw::geom::degToRad(meta->get<double>("RA_DEG"));
-        dec = lsst::afw::geom::degToRad(meta->get<double>("DEC_DEG"));
+    double latitude = visitInfo->getObservatory().getLatitude();
+    double lst_obs = visitInfo->getEra();
+    double ra = visitInfo->getBoresightRaDec().getRa();
+    double dec = visitInfo->getBoresightRaDec().getDec();
+    double hourAngle = visitInfo->getBoresightHourAngle();
+    airMass = visitInfo->getBoresightAirmass();
+    mjd = visitInfo->getDate().get(lsst::daf::base::DateTime::MJD);
+
+    // lsstSim doesn't manage ERA (and thus Hour Angle) properly, so it's going to be NaN.
+    // Because we need the refraction vector later, go with 0 HA to prevent crashes on that NaN.
+    if (std::isnan(hourAngle) == true) {
+        hourAngle = 0;
```

Comment:
This is a known problem with lsstSim: it will happen with all lsstSim data. I'm not sure it's worth cluttering the logs with a known issue?